File: RepoTransBench/target_projects/Python/C++/initstring_linkedin2username/public_tests/test_public_namemutator.cpp
#include <gtest/gtest.h>
#include <string>
#include <map>
#include <vector>
#include "linkedin2username.h"

struct PubNameTestCase {
    std::string input_name;
    std::map<std::string, std::string> expected;
};

class PublicNameMutatorParamTest : public testing::TestWithParam<PubNameTestCase> {};

TEST_P(PublicNameMutatorParamTest, CleanAndSplitNamePublic) {
    auto param = GetParam();
    NameMutator nm(param.input_name);
    EXPECT_EQ(nm.name, param.expected);
}

INSTANTIATE_TEST_SUITE_P(
    PublicNameMutatorParamTests,
    PublicNameMutatorParamTest,
    testing::Values(
        PubNameTestCase{"Sam Lee", {{"first", "sam"}, {"last", "lee"}, {"second", ""}}},
        PubNameTestCase{"Ms. Eva O'Brien", {{"first", "eva"}, {"last", "obrien"}, {"second", ""}}},
        PubNameTestCase{"Prof. Łukasz Nowak (PhD)", {{"first", "lukasz"}, {"last", "nowak"}, {"second", ""}}},
        PubNameTestCase{"María-José Carreño", {{"first", "maria"}, {"last", "carreno"}, {"second", ""}}},
        PubNameTestCase{"Chris (CEO) Smithers", {{"first", "chris"}, {"last", "smithers"}, {"second", ""}}}
    )
);

File: RepoTransBench/target_projects/Python/Rust/initstring_linkedin2username/public_tests/test_public_namemutator.rs
use std::collections::HashMap;
use linkedin2username::NameMutator;

fn mkmap(first: &str, last: &str, second: &str) -> HashMap<String, String> {
    let mut h = HashMap::new();
    h.insert("first".to_string(), first.to_string());
    h.insert("last".to_string(), last.to_string());
    h.insert("second".to_string(), second.to_string());
    h
}

#[test]
fn test_clean_and_split_name_public() {
    let samples = vec![
        ("Sam Lee", mkmap("sam", "lee", "")),
        ("Ms. Eva O'Brien", mkmap("eva", "obrien", "")),
        ("Prof. Łukasz Nowak (PhD)", mkmap("lukasz", "nowak", "")),
        ("María-José Carreño", mkmap("maria", "carreno", "")),
        ("Chris (CEO) Smithers", mkmap("chris", "smithers", "")),
    ];
    for (input, expected) in samples {
        let nm = NameMutator::new(input);
        assert_eq!(nm.name, expected);
    }
}